internal/database/postgres.go

```diff
@@ -720,3 +720,148 @@ func (conn PostgresConn) RemoveFile(remove broadcast.RemoveFile) error {
 	}
 	return err
 }
+
+func (conn PostgresConn) HistoricalData(hostname string) (broadcast.HistoricalData, error) {
+	data := broadcast.HistoricalData{}
+	samples, err := conn.db.Query(`SELECT
+		s.Gpu,
+		s.Received,
+		s.MemoryUtilisation,
+		s.GpuUtilisation,
+		s.MemoryUsed,
+		s.FanSpeed,
+		s.Temp,
+		s.MemoryTemp,
+		s.GraphicsVoltage,
+		s.PowerDraw,
+		s.GraphicsClock,
+		s.MaxGraphicsClock,
+		s.MemoryClock,
+		s.MaxMemoryClock,
+		s.InUse,
+		s.UserName
+		FROM Stats s
+		INNER JOIN GPUs g ON g.Uuid = s.Gpu
+		WHERE g.Machine=$1
+		ORDER BY s.Received`,
```

Comment:
For a machine with more than one GPU, how are we enforcing an ordering that alternates between GPUs in the same pattern? If I'm reading this right, I think we could potentially get the data out the database in the order `GPU1 @ 1s, GPU2 @ 1s, GPU2 @ 2s, GPU1 @ 2s, ...`.

It should be enough to make this line `ORDER BY s.Recieved, s.Gpu`

Reply:
we dont need to do that, because the output is agnostic of the order (confirm @NathanielB123 ?)

Reply:
> we dont need to do that, because the output is agnostic of the order (confirm @NathanielB123 ?)

I don't think this is true? I draw a line for each gpu, so if gpu order swaps, lines will use the wrong points

Reply:
currently the data is given out in `[(Timestamp, [GPU])]`. Does the ordering of `[GPU]` matter? if so, this needs to be fixed. If not, then things should work out alright

Reply:
Yes, the ordering does matter

Reply:
oh okay, then this needs to be fixed as per @danielg0's comment